src/cpp/src/image_generation/models/sd3_transformer_2d_model.cpp

```diff
@@ -77,54 +79,37 @@ SD3Transformer2DModel& SD3Transformer2DModel::reshape(int batch_size,
     height /= m_vae_scale_factor;
     width /= m_vae_scale_factor;
 
-    std::map<std::string, ov::PartialShape> name_to_shape;
-
-    for (auto&& input : m_model->inputs()) {
-        std::string input_name = input.get_any_name();
-        name_to_shape[input_name] = input.get_partial_shape();
-        if (input_name == "timestep") {
-            name_to_shape[input_name][0] = 1;
-        } else if (input_name == "hidden_states") {
-            name_to_shape[input_name] = {batch_size, name_to_shape[input_name][1], height, width};
-        } else if (input_name == "encoder_hidden_states") {
-            name_to_shape[input_name][0] = batch_size;
-            name_to_shape[input_name][1] =
-                tokenizer_model_max_length *
-                2;  // x2 is necessary because of the concatenation of prompt_embeds and t5_prompt_embeds
-        } else if (input_name == "pooled_projections") {
-            name_to_shape[input_name][0] = batch_size;
-        }
-    }
-
-    m_model->reshape(name_to_shape);
+    SD3Transformer2DModel::Inference::reshape(m_model, batch_size, height, width, tokenizer_model_max_length);
 
     return *this;
 }
 
 SD3Transformer2DModel& SD3Transformer2DModel::compile(const std::string& device, const ov::AnyMap& properties) {
     OPENVINO_ASSERT(m_model, "Model has been already compiled. Cannot re-compile already compiled model");
-    ov::CompiledModel compiled_model = utils::singleton_core().compile_model(m_model, device, properties);
-    ov::genai::utils::print_compiled_model_properties(compiled_model, "SD3 Transformer 2D model");
-    m_request = compiled_model.create_infer_request();
+
+    if (device == "NPU") {
```

Comment:
```suggestion
    if (device.find("NPU") != std::string::npos) {
```
because NPU can be with several tiles named as "NPU.1", etc